src/imitation/scripts/train_adversarial.py

```diff
@@ -159,12 +151,13 @@ def train(_run,
       visualizer = None
 
     # Main training loop.
+    n_epochs = total_timesteps // trainer.gen_batch_size
+
     for epoch in tqdm.tqdm(range(1, n_epochs+1), desc="epoch"):
```

Comment:
Why not use `trainer.train`?

Reply:
`visualizer` needs two callbacks, one after `train_gen` and another after `train_disc`. I suppose I could merge these two blocks of code by adding these two callbacks into `train()`?

Also would add the `save(trainer, ...` to this hypothetical callback.

Reply:
Another way to merge this code with `train()` would be to move the `Visualizer` into `Adversarial` and use a callback for the checkpointing `save(trainer, ...`.

Reply:
I got rid of `AdversarialTrainer.train` in favor of using this loop (alternative was to use multiple callbacks in `AdversarialTrainer.train()`)